src/components/Button/Button.jsx

```diff
@@ -2,31 +2,50 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './Button.scss';
 import classnames from 'clsx';
+import {Typography} from '../Typography';
 
-export const ButtonSizes = ['small', 'medium', 'big'];
-export const ButtonVariants = ['default', 'primary', 'ghost'];
-export const ButtonColors = ['accent', 'inherit', 'success', 'warning', 'danger', 'reverse'];
+export const ButtonSizes = ['small', 'default', 'big'];
+export const ButtonVariants = ['default', 'ghost', 'outlined'];
+export const ButtonColors = ['default', 'accent', 'success', 'warning', 'danger'];
 
-export const Button = ({label, onClick, size, isDisabled, icon, variant, color, ...props}) => {
+export const Button = ({label, onClick, size, isReversed, isDisabled, icon, variant, color, className, ...props}) => {
     return (
         <button
-            className={classnames(styles.button, styles[variant], styles[color], (icon === null) ? null : styles.icon)}
+            className={
+                classnames(
+                    styles.button,
+                    styles[`size_${size}`],
+                    styles[`variant_${variant}`],
+                    styles[`color_${color}`],
+                    {[styles.icon]: (icon && label)},
+                    {[styles['icon-button']]: !label},
+                    {[styles.reverse]: isReversed},
+                    className
+                )
+            }
             type="button"
+            disabled={isDisabled}
             onClick={onClick}
             {...props}
         >
-            <i>{icon}</i>{label}
+            {icon && <icon.type {...icon.props} size={(size === 'big') ? 'default' : size}/>}
+            {label &&
+                <Typography isNowrap component="span" variant="button" isUpperCase={size === 'big'} weight={(size === 'small') ? 'light' : 'default'}>
+                    {label}
+                </Typography>}
         </button>
     );
 };
 
 Button.defaultProps = {
```

Comment:
What about having a default value for onClick ?

Reply:
As `onClick` is required, do really need a default value?

Reply:
Would simplify the writing of unit test at least :-)

Reply:
I don't know, I would say no